frb_codegen/src/generator/dart/ty_struct.rs

```diff
@@ -120,18 +125,26 @@ impl TypeDartGeneratorTrait for TypeStructRefGenerator<'_> {
                 .collect::<Vec<_>>();
             if has_methods {
                 constructor_params.insert(0, extra_argument);
+                if self.ir.is_exception {
+                    constructor_params.push("Backtrace? backtrace".to_string());
+                }
             }
             let constructor_params = constructor_params.join("");
 
             format!(
-                "{}{}class {} with _${} {{
+                "{}{}class {} with _${} {} {{
                 const factory {}({{{}}}) = _{};
                 {}
             }}",
                 comments,
                 metadata,
                 self.ir.name,
                 self.ir.name,
+                if self.ir.is_exception {
+                    "implements Exception"
```

Comment:
Btw as we have discussed earlier it should be `FrbException` not `Exception` I guess. That is also why we should extract the duplicated code into a new function - we will never forget to modify all changes